### ISSUE TITLE
Cannot find org.atmosphere.play.Router

### ISSUE THREAD

**Issue by bb-work** [Atmosphere/atmosphere-play#39]
Compilation fails for Play! chat sample as downloaded from atmosphere-samples site, no modifications.  Using Play! 2.5.4

bba0124$ activator compile
[info] Loading project definition from /Users/bba0124/Downloads/atmosphere-samples-master/play-samples/chat/project
[info] Set current project to chat (in build file:/Users/bba0124/Downloads/atmosphere-samples-master/play-samples/chat/)
[info] Compiling 3 Scala sources and 7 Java sources to /Users/bba0124/Downloads/atmosphere-samples-master/play-samples/chat/target/scala-2.11/classes...
[error] /Users/bba0124/Downloads/atmosphere-samples-master/play-samples/chat/app/Global.java:21: error: cannot find symbol
[error] import org.atmosphere.play.Router;
[error]                           ^
[error]   symbol:   class Router
[error]   location: package org.atmosphere.play
[error] /Users/bba0124/Downloads/atmosphere-samples-master/play-samples/chat/app/Global.java:40: error: cannot find symbol
[error]         return Router.dispatch(request);
[error]                ^
[error]   symbol:   variable Router
[error]   location: class Global
[error] 2 errors
[error] (compile:compile) javac returned nonzero exit code
[error] Total time: 5 s, completed Jul 12, 2016 2:06:17 PM

**Comment by bb-work** [comment]
So I eventualy deduced that this was a deprecated function.  Still, updating your examples would be useful!!

**Comment by jfarcand** [comment]
Pull request welcomed!

**Comment by bb-work** [comment]
If I could actually figure out how to make this stuff work, maybe I would.  It would help to have an actual working example, though!

**Comment by jfarcand** [comment]
I understand...but I don't have the cycle to help right now. Maybe @stegmann-idnow

**Comment by stegmann-idnow** [comment]
Hey, the play's GlobalSettings class is deprecated and will be removed. 
For play 2.5 you have to define a request handler instead. 

If you configure the following in the application.conf it should work:
play.http.requestHandler = "org.atmosphere.play.AtmosphereHttpRequestHandler"

And of course you have to remove the onRouteRequest method in your Global.java. 

I try to migrate the chat example in the next days.

**Comment by bb-work** [comment]
OK, that got rid of the compilation error, but the web socket doesn't connect:

WebSocket connection to 'ws://localhost:9000/chat?X-Atmosphere-tracking-id=0&X-Atmosphere-Framework=2.3.2-javascript&X-Atmosphere-Transport=websocket&X-Atmosphere-TrackMessageSize=true&Content-Type=application/json&X-atmo-protocol=true' failed: Error during WebSocket handshake: Unexpected response code: 404

**Comment by DeveloperInResidence** [comment]
I'm having the same problem, though I'm using HTTPS, so I've added play.http**s**.requestHandler = "org.atmosphere.play.AtmosphereHttpRequestHandler" to my application.conf.
Still, I get a 404 error when I try to connect (trying websockets first, falling back on long-polling).
Has anyone been able to use the ManagedService with Play at all?

**Comment by bb-work** [comment]
Yes, I have.

My conf file includes:


play.http.requestHandler = "org.atmosphere.play.AtmosphereHttpRequestHandler"


My main application looks like:

import static org.atmosphere.play.AtmosphereCoordinator.instance;

public class Application extends Controller {



    @Inject

    public Application(ApplicationLifecycle lifecycle){



        instance.discover(SelectPolicyHandler.class).ready();


        logger.info("Creating Application");


        lifecycle.addStopHook(() -> {

            instance().shutdown();

            return CompletableFuture.completedFuture(null);

        });

    }


    public Result index() {

        return ok(index.render());

    }



}

SelectPolicyHandler is the class with ManagedService.

**Comment by DeveloperInResidence** [comment]
Thank you! The instance.discover part in the controller solved that problem for me.
Now I've got a different problem which I can't find any good solution for, but perhaps you have encountered when using Play?

I'm trying to authenticate access to the ManagedService, but the Authenticated annotation doesn't seem to work on methods in that class, nor can I access the Play session.
I've tried enabling the Play session by using a class implementing the AtmospherePlaySessionConverter, but due to classloading issues it appears it cannot be used in development mode. Have you perhaps found a way to do authentication or access the Play session in development mode?
